src/lib/bootloader/grub2bls.rb

```diff
@@ -0,0 +1,193 @@
+
+    # @return true if configuration is already proposed
+    def proposed?
+      @is_proposed
+    end
+
+    # writes configuration to target disk
+    def write(*)
+      install_bootloader if Yast::Stage.initial # while new installation only (currently)
+      create_menu_entries
+      install_bootloader
+      @sections.write
+      write_menu_timeout
+      # writing kernel parameter to /etc/kernel/cmdline
+      File.open(File.join(Yast::Installation.destdir, CMDLINE), "w+") do |fw|
+        fw.puts(grub_default.kernel_params.serialize)
+      end
+    end
+
+    # merges other bootloader configuration into this one.
+    # It have to be same bootloader type.
+    # rubocop:disable Metrics/AbcSize
+    def merge(other)
+      raise "Invalid merge argument #{other.name} for #{name}" if name != other.name
+
+      log.info "merging: timeout: #{grub_default.timeout}=>#{other.grub_default.timeout}"
+      log.info "         mitigations: #{cpu_mitigations.to_human_string}=>" \
+               "#{other.cpu_mitigations.to_human_string}"
+      log.info "         grub_default.kernel_params: #{grub_default.kernel_params.serialize}=>" \
+               "#{other.grub_default.kernel_params.serialize}"
+
+      merge_sections(other)
+      merge_grub_default(other)
+
+      log.info "merging result: timeout: #{grub_default.timeout}"
+      log.info "                mitigations: #{cpu_mitigations.to_human_string}"
+      log.info "                kernel_params: #{grub_default.kernel_params.serialize}"
+    end
+    # rubocop:enable Metrics/AbcSize
+
+    # @return [Array<String>] packages required to configure given bootloader
+    def packages
+      res = super
+      res << ("grub2-" + Yast::Arch.architecture + "-efi-bls")
+      res << "sdbootutil"
+      res << "grub2"
+      res
+    end
+
+  private
+
+    SDBOOTUTIL = "/usr/bin/sdbootutil"
+    OS_RELEASE_PATH = "/etc/os-release"
+
+    def grubenv_path
+      str = Yast::Misc.CustomSysconfigRead("ID_LIKE", "openSUSE",
+        OS_RELEASE_PATH)
+      os = str.split.first
+      File.join("/boot/efi/EFI/", os, "/grubenv")
+    end
+
+    # @return [String] return default boot as string or "" if not set
+    # or something goes wrong
+    def read_menu_timeout
+      grub_default.timeout = Yast::Misc.CustomSysconfigRead("timeout", "",
+        grubenv_path)
+      log.info "Boot timeout: #{grub_default.timeout}"
+    end
+
+    def write_menu_timeout
+      ret = Yast::Execute.on_target(SDBOOTUTIL,
+        "set-timeout",
+        grub_default.timeout,
+        allowed_exitstatus: [0, 1])
+
+      return unless ret != 0
+
+      # fallback directly over grub2-editenv
+      Yast::Execute.on_target("/usr/bin/grub2-editenv", grubenv_path,
+        "set", "timeout=#{grub_default.timeout}")
+    end
+
+    def merge_sections(other)
+      return if !other.sections.default || other.sections.default.empty?
+
+      @sections.default = other.sections.default
+    end
+
+    def create_menu_entries
```

Comment:
Most of this code can be shared to the systemd-boot case, as they are using the same interface (sdbootutil).  Would make case to unify it?

Reply:
I have moved all sdbootutil calls to one class bls.rb